File: src/main/java/com/oshi/libgenericsettings/data/ExpandableTitleSubtitleCheckableItemsData.java
package com.oshi.libgenericsettings.data;


import android.support.annotation.ColorRes;
import android.support.annotation.NonNull;

import java.util.List;

public class ExpandableTitleSubtitleCheckableItemsData extends ExpandableTitleCheckableItemsData {

    protected String subtitle;
    protected @ColorRes int subtitleColor;

    public ExpandableTitleSubtitleCheckableItemsData(@NonNull String title, @NonNull String subtitle, @NonNull List<CheckableSubItem> items) {
        super(title, items);
        this.subtitle = subtitle;
    }

    public String getSubtitle() {
        return subtitle;
    }

    public void setSubtitle(String subtitle) {
        this.subtitle = subtitle;
    }

    public int getSubtitleColor() {
        return subtitleColor;
    }

    public void setSubtitleColor(int subtitleColor) {
        this.subtitleColor = subtitleColor;
    }

    @Override
    public int getViewType() {
        return VIEW_TYPE_SETTINGS_EXPANDABLE_TITLE_SUBTITLE_CHECKABLE_ITEMS;
    }
}

File: src/main/java/com/oshi/libgenericsettings/data/ExpandableTitleSubtitleCheckableItemsData.kt
package com.oshi.libgenericsettings.data

import android.support.annotation.ColorRes

/**
 * Expandable Title & Subtitle with Checkables sub items. Being use in layout/view_type_expandable_title_subtitle_checkable_items.xml
 */
class ExpandableTitleSubtitleCheckableItemsData(title : String, var subtitle : String, items : List<CheckableSubItem>) : ExpandableTitleCheckableItemsData(title, items) {

    @ColorRes var subtitleColor : Int = 0

    override fun getViewType(): Int {
        return VIEW_TYPE_SETTINGS_EXPANDABLE_TITLE_SUBTITLE_CHECKABLE_ITEMS
    }

}